src/test/java/org/elasticsearch/ingest/openshift/OpenshiftIndicesUtilTests.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 Lukáš Vlček
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.elasticsearch.ingest.openshift;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.metadata.AliasMetaData;
+import org.elasticsearch.cluster.metadata.AliasOrIndex;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static org.elasticsearch.ingest.openshift.OpenshiftIndicesUtil.generateInitialIndexName;
+import static org.elasticsearch.ingest.openshift.OpenshiftIndicesUtil.generateWriteAliasName;
+import static org.elasticsearch.ingest.openshift.OpenshiftIndicesUtil.hasDataModelPrefix;
+
+public class OpenshiftIndicesUtilTests extends ESTestCase {
+
+    public void testGenerateInitialIndexName() {
+        assertEquals("app-000001", generateInitialIndexName("app-write"));
+        assertEquals("alias-write-000001", generateInitialIndexName("alias-write-write"));
+        assertEquals("-000001", generateInitialIndexName("-write"));
+    }
+
+    public void testGenerateWriteAliasName() {
+        assertEquals("app-write", generateWriteAliasName("app-000001"));
+        assertEquals("foo-write-write", generateWriteAliasName("foo-write-000001"));
+        assertEquals("-write", generateWriteAliasName("-000001"));
+    }
+
+    public void testDataModelPrefix() {
+        for (String prefix : Arrays.asList("app", "infra", "audit")) {
+
+            assertFalse(hasDataModelPrefix("." + prefix));
+            assertFalse(hasDataModelPrefix(prefix));
```

Comment:
is the reason we don't want this to be seen as the data model because we currently have our "read" aliases match this?

Reply:
No, it has to do more with question whether we consider the comma to be part of our data model naming schema.
In other words, do we care if the index or alias starts with `app` or `app-` ?

I did not find any reason to exclude the comma from the naming schema.
One reason: it is edge case but it looks more robust to me because it will not include name `apple` into schema by mistake. (No pun against $AAPL intended)